HW03/ihar_shkuryn/ihar_shkuryn_hw_03_t_02.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class Homework3
+  PATTERN = %r{(?:[0-9]{1,3}\.){3}[0-9]{1,3} - - \[\d{2}+/\w{3}/\d{4}:\d{2}:\d{2}:\d{2} \+\d{4}\]}
+  REG_IP = /(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})/
+  REG_DATETIME = %r{(\[\d+/\w+/\d+:\d+:\d+:\d+ \+\d{4}\])}
+  REG_TEXT = /(".+")/
+
+  def task2(log)
```

Comment:
You have logs
Split logs into lines
Select all rows that include a POST request
Use map for the selected results and do something like this in the map block:

datetime = line.scan(DATETIME).first
address = line.scan(ADDRESS).first.upcase
ip = line.scan(IP).first
"#{datetime} FROM: #{ip} TO: #{address}"